core/src/test/java/com/rudderstack/core/CentralPluginChainTest.kt

```diff
@@ -16,132 +16,109 @@ package com.rudderstack.core
 
 import com.rudderstack.core.internal.CentralPluginChain
 import com.rudderstack.models.Message
-import com.rudderstack.models.TrackMessage
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
 import org.hamcrest.MatcherAssert.assertThat
 import org.hamcrest.Matchers.equalTo
 import org.hamcrest.Matchers.`is`
-import org.hamcrest.Matchers.not
 import org.junit.Before
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.mockito.InjectMocks
-import org.mockito.Mock
-import org.mockito.junit.MockitoJUnitRunner
-import org.mockito.kotlin.any
-import org.mockito.kotlin.argumentCaptor
-import org.mockito.kotlin.mock
-import org.mockito.kotlin.verify
-import org.mockito.kotlin.whenever
-import java.util.*
-import java.util.concurrent.atomic.AtomicInteger
+
+import io.mockk.mockk
+import io.mockk.slot
 
 /**
  * Testing flow of control through plugins.
  * We will check if plugins order is maintained.
  * Avoiding spying of objects.
  *
  */
-@RunWith(MockitoJUnitRunner::class)
 class CentralPluginChainTest {
 
-    companion object{
-        //Plugins should be called in this order
-        private const val SUB_PLUGIN_1_1_SL_NO = 1
-        private const val SUB_PLUGIN_2_1_SL_NO = 2
-        private const val MAIN_PLUGIN_1_SL_NO = 3
-        private const val SUB_PLUGIN_1_2_SL_NO = 4
-        private const val MAIN_PLUGIN_2_SL_NO = 5
-        private const val MAIN_PLUGIN_3_SL_NO = 6
-        private const val TOTAL_NO_OF_CALLS = 7
-    }
-    @Mock
+    @MockK
     lateinit var mockPlugin1: Plugin
 
-    @Mock
+    @MockK
     lateinit var mockPlugin2: Plugin
 
-    @Mock
+    @MockK
```

Comment:
DestinationPlugins are interfaces, why do we need mockK?

Reply:
I tried to remove the mockk and the tests started failing. Even in the current feat/android branch, I tried to remove @Mock and the same result, the test failed again.